src/plugins/Tus.js

```diff
@@ -166,7 +166,17 @@ module.exports = class Tus extends Plugin {
         this.resetUploaderReferences(file.id)
         resolve(upload)
       }
-      optsTus.metadata = file.meta
+
+      // tusd uses metadata field "filetype", not "type"
+      const meta = Object.assign({}, file.meta)
+      if (
+        Object.prototype.hasOwnProperty.call(meta, 'type') &&
+        !Object.prototype.hasOwnProperty.call(meta, 'filetype')
+      ) {
+        meta.filetype = meta.type
+        delete meta.type
+      }
+      optsTus.metadata = meta
 
```

Comment:
Feels like this complicates everything a bit. If we just add `filetype` (maybe check if it is not already set), it would be simpler.

Also, could you add `filename`? See https://github.com/tus/tus-js-client/commit/ebc5189eac35956c9f975ead26de90c896dbe360 and https://github.com/tus/tusd/blob/5b376141903c1fd64480c06dde3dfe61d191e53d/unrouted_handler.go#L640-L643